lib/os/Kconfig.cbprintf

```diff
@@ -3,6 +3,7 @@
 
 choice CBPRINTF_IMPLEMENTATION
 	prompt "Capabilities of cbprintf implementation"
+	depends on !PICOLIBC
```

Comment:
This should not be on the choice, but on the config items.

Reason, if someone elsewhere (downstream) want to extend `CBPRINTF_IMPLEMENTATION`, they can do so by writing:
```
choice CBPRINTF_IMPLEMENTATION
     prompt "Downstream cbprintf implementation"

config DOWNSTREAM_CBPRINTF
    bool "my impl"

endchoice
```

in such a case `CBPRINTF_COMPLETE` will suddenly reappear, which is clearly wrong.
```suggestion
```

Reply:
I don't understand what change you're suggesting be made here.

Reply:
fwiw, these comments relate to changes in this PR, not #39563